runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -616,13 +626,20 @@ public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V>
      * @param value         The value to unmap.
      */
     protected void unmapSecondaryIndexes(K key, V value) {
-        if (value != null) {
-            final Map.Entry<K, V> previousEntry
-                    = new AbstractMap.SimpleImmutableEntry<>(key, value);
-            indexFunctions.parallelStream()
-                    .forEach(f -> f.getIndexFunction().generateIndex(key, value)
-                            .forEach(i -> indexMap.get(f).remove((I) i, previousEntry)));
+        try {
+            if (value != null) {
+                final Map.Entry<K, V> previousEntry
+                        = new AbstractMap.SimpleImmutableEntry<>(key, value);
+                indexFunctions.parallelStream()
+                        .forEach(f -> f.getIndexFunction().generateIndex(key, value)
+                                .forEach(i -> indexMap.get(f).remove((I) i, previousEntry)));
 
+            }
+        } catch (Exception e) {
```

Comment:
Can we also flush or dump the index data ? To save on memory. 

Another thing, it is hard to differentiate between a failed index and a table which had no index registered if we miss the error message for some reason. It it possible to remember that at least while this Table is in memory.